connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala

```diff
@@ -163,13 +165,106 @@ class StreamingQuerySuite extends RemoteSparkSession with SQLHelper {
       assert(!terminated)
 
       q.stop()
-      // TODO (SPARK-43032): uncomment below
-      // eventually(timeout(1.minute)) {
-      // q.awaitTermination()
-      // }
+      eventually(timeout(1.minute)) {
+        q.awaitTermination()
+      }
+    }
+  }
+
+  class TestForeachWriter[T] extends ForeachWriter[T] {
```

Comment:
Can you move this class outside of the test class e.g.
```
class XXSuite {}
class TestForeachWriter[T] ... r{}
```
And see if the serdes problem is resolved.

Reply:
Thanks! That worked!